api/server/secrets.go

```diff
@@ -0,0 +1,138 @@
+package server
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/libopenstorage/openstorage/secrets"
+)
+
+// TODO: Add swagger yaml
+func (c *clusterApi) setDefaultSecretKey(w http.ResponseWriter, r *http.Request) {
+
+	method := "setDefaultSecretKey"
+	var secReq secrets.DefaultSecretKeyRequest
+
+	if err := json.NewDecoder(r.Body).Decode(&secReq); err != nil {
+		c.sendError(c.name, method, w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	err := c.SecretManager.SecretSetDefaultSecretKey(
+		secReq.DefaultSecretKey,
+		secReq.Override,
+	)
+
+	if err != nil {
+		c.sendError(c.name, method, w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+}
+
+// TODO: Add swagger yaml
+func (c *clusterApi) getDefaultSecretKey(w http.ResponseWriter, r *http.Request) {
+
+	method := "getDefaultSecretKey"
+
+	secretValue, err := c.SecretManager.SecretGetDefaultSecretKey()
+	if err != nil {
+		c.sendError(c.name, method, w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	secResp := &secrets.GetSecretResponse{
+		SecretValue: secretValue,
+	}
+	json.NewEncoder(w).Encode(secResp)
+}
+
+// TODO: Add swagger yaml
+func (c *clusterApi) secretsLogin(w http.ResponseWriter, r *http.Request) {
+	var secReq secrets.SecretLoginRequest
+	method := "secretsLogin"
+
+	if err := json.NewDecoder(r.Body).Decode(&secReq); err != nil {
+		if err == secrets.ErrNotAuthenticated {
+			c.sendError(c.name, method, w, err.Error(), http.StatusForbidden)
```

Comment:
should return a StatusUnauthorized - 401

Reply:
updated